### PR TITLE
Add description and enable schema linter with exclusions

### PR DESCRIPTION
**Pull request by markkelnar**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Identify and add a few remaining root query descriptions.
Enable the schema linter but exclude a few types of linting errors that we still have left to remedy.

Does this close any currently open issues?
------------------------------------------
#1790 
https://github.com/wp-graphql/wp-graphql/pull/1829

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System:** …

**WordPress Version:** …

### CODE REVIEW

**Comment by szepeviktor** [comment]
Please reconsider removing `|| true` as `--except` does not display errors.
Hidden errors are the worst things.

**Comment by szepeviktor** [comment]
You know what!
Add this command.

```bash
graphql-schema-linter --except=relay-connection-types-spec /tmp/schema.graphql || true
```

**Comment by szepeviktor** [comment]
This ignores specific rules in specific objects.
```
--ignore '{"relay-page-info-spec":["PageInfo"],"defined-types-are-used":["MenuItemsWhereArgs","PostObjectUnion","TermObjectUnion","TimezoneEnum"]}'
```

File: src/Registry/TypeRegistry.php [hunk 1]
```diff
@@ -569,8 +569,9 @@ public function init_type_registry( TypeRegistry $type_registry ) {
 					'RootQuery',
 					$group_name . 'Settings',
 					[
-						'type'    => ucfirst( $group_name ) . 'Settings',
-						'resolve' => function() use ( $setting_type ) {
+						'type'        => ucfirst( $group_name ) . 'Settings',
+						'description' => sprintf( __( 'Fields of the \'%s\' settings group', 'wp-graphql' ), $group_name ),
```
Comment (szepeviktor) [inline src/Registry/TypeRegistry.php:573]:
```suggestion
						'description' => sprintf( __( "Fields of the '%s' settings group", 'wp-graphql' ), $group_name ),
```

File: .github/workflows/schema-linter.yml [hunk 2]
```diff
@@ -45,10 +45,10 @@ jobs:
...
       - name: Generate the Static Schema
-        # 💡 Neutralize for now
+        # TODO: fix remaining exceptions
         run: |
           cd /tmp/wordpress/
           # Output: /tmp/schema.graphql
           wp graphql generate-static-schema
           cd /tmp/wordpress/wp-content/plugins/wp-graphql/
-          graphql-schema-linter /tmp/schema.graphql || true
+          graphql-schema-linter --ignore '{"relay-page-info-spec":["PageInfo"],"defined-types-are-used":["MenuItemsWhereArgs","PostObjectUnion","TermObjectUnion","TimezoneEnum"]}' /tmp/schema.graphql
```
Comment (szepeviktor) [inline .github/workflows/schema-linter.yml:54]:
```suggestion
          graphql-schema-linter --except=relay-connection-types-spec --ignore '{"relay-page-info-spec":["PageInfo"],"defined-types-are-used":["MenuItemsWhereArgs","PostObjectUnion","TermObjectUnion","TimezoneEnum"]}' /tmp/schema.graphql
```

If you exclude `relay-connection-types-spec` the rule globally - as discussed in the top comment - we could see the remaining problems.

Comment (szepeviktor) [reply .github/workflows/schema-linter.yml:54]: Maybe only two!!

**Comment by szepeviktor** [comment]
Another thing.
We have `"relay-page-info-spec":["PageInfo"]` but this error shows up.

```
1:1 A `PageInfo` object type is required as per the Relay spec.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  relay-page-info-spec
```

Could it be we need `"relay-page-info-spec":["Relay"]` ?

**Comment by markkelnar** [comment]
```
12044:1 The `UserToPageConnection.pageInfo` field must return a non-null `PageInfo` object not `WPPageInfo`                                                                                                                                                                                                                                                                                                                                                                                                                                                          relay-connection-types-spec      
```

I've tried different combinations for the ignore `relay-connection-types-spec`, but can only seem to get it to work if I specify each of the 67 group names, like `TermNodeToEnqueuedScriptConnection`.

**Comment by szepeviktor** [comment]
> I've tried different combinations for the ignore `relay-connection-types-spec`, but can only seem to get it to work if I specify each of the 67 group names, like `UserToPageConnection`.

It is possible with `--except=relay-connection-types-spec`, please see the above suggestion.

File: .github/workflows/schema-linter.yml (hunk 2 above)
Comment (markkelnar) [reply .github/workflows/schema-linter.yml:54]: Here's how I've been running the linter locally and testing the rules.

```
# Install node graphql linter
npm install graphql-schema-linter

# Start the WP app
# Verify the WP is up by visiting http://localhost:8091/
composer run-app

# Generate the schema in the WP, copy the file to an accessible location, then run the node linter command
docker-compose exec -- app /usr/local/bin/wp --allow-root graphql generate-static-schema
docker-compose exec -- app cp /tmp/schema.graphql /var/log/apache2/
node_modules/.bin/graphql-schema-linter .log/app/schema.graphql
```

Comment (szepeviktor) [reply .github/workflows/schema-linter.yml:54]: Okay. Just add `--except` and `--ignore`.

**Comment by markkelnar** [comment]
Maybe I go back to my [earlier commit](https://github.com/wp-graphql/wp-graphql/pull/1851/commits/02f79adde212484fbeb07132ba339f2fba695f87#diff-2f6413f669865592eff59ee2a55f09371bfcdca46e6b2af84dfa80cc8b16d6b1L54) using --except and not a combination of --except and --ignore.

**Comment by szepeviktor** [comment]
What a green outcome! 🍏 

Could you add `graphql-schema-linter /tmp/schema.graphql || true` to see what ignored rules have to say?

**Comment by szepeviktor** [comment]
Thank you.

**Comment by jasonbahl** [comment]
@szepeviktor @markkelnar thanks so much for the work on this one! 

I added the 2nd command, per @szepeviktor suggestion, so now the linter runs and will pass/fail with the rules we want to ignore, but will run again and output the errors for the rules we're choosing to ignore, so we can still easily see them when we want to revisit and fix them. 

Output now looks like below, where first run succeeds with no errors, 2nd run shows the errors but doesn't fail the build: 

Screen Shot 2021-04-12 at 10 34 07 PM

**Pull request merged**